src/DelegateCompensationStaker.sol

```diff
@@ -44,37 +46,58 @@ abstract contract DelegateCompensationStaker is Staker {
   /// model.
   error DelegateCompensation__MethodNotSupported();
 
-  /// @notice Tracks whether a delegate has already been initialized for compensation.
-  mapping(address delegate => bool isInitialized) public delegateInitialized;
+  /// @notice A mapping from a delegate's address to their unique compensation deposit identifier.
+  /// @dev This allows for efficient lookup of a delegate's deposit information. A return value of 0
+  /// indicates that the delegate is uninitialized.
+  mapping(address delegate => DepositIdentifier) public delegateDepositId;
+
+  /// @param _rewardToken ERC20 token in which rewards will be denominated.
+  /// @param _stakeToken Delegable governance token (unused in delegate compensation but required by
+  /// base Staker).
+  /// @param _earningPowerCalculator The contract that will serve as the initial calculator of
+  /// earning power for the staker system.
+  /// @param _maxBumpTip The maximum tip that can be paid to a bumper for updating earning power.
+  /// @param _admin Address which will have permission to manage reward notifiers, claim fee
+  /// parameters, the max bump tip, and the reward calculator.
+  constructor(
+    IERC20 _rewardToken,
+    IERC20 _stakeToken,
```

Comment:
I don't think we need to accept this as a constructor arg since it isn't being used. Could be simply pass in the 0 address on line 68?

Reply:
Also, since we have a constructor now is there anything holding us back from removing abstract from the contract?